types/src/main.ts

```diff
@@ -1341,6 +1350,83 @@ export interface WorkspaceSymbolParams {
 	query: string;
 }
 
+/**
+ * The kind of a code action.
+ *
+ * Kinds are a hierarchical list of identifiers separated by `.`, e.g. `"refactor.extract.function"`.
+ *
+ * The set of kinds is open and client needs to announce the kinds it supports to the server during
+ * initialization.
+ */
+export type CodeActionKind = string;
+
+/**
+ * A set of predefined code action kinds
+ */
+export namespace CodeActionKind {
```

Comment:
There is no `Empty` type but VS Code does have one. Is this intentional?

Reply:
I though that this is not necessary in the LSP since we can simply use the empty string. In VSCode it is a class with a private constructor hence you can only make new kinds by starting with Empty and then append. Since we have literals I felt this is overkill and left it out.